### PR TITLE
Fix feature sign in

### PR DESCRIPTION
**Pull request by jpascual1994**
This PR includes:
- Show sign in errors and stay in the same page

Trello card: https://trello.com/c/nbO6LPq0/15-3-as-a-user-i-should-be-able-to-login-so-i-can-use-my-target-mvd-account

### CODE REVIEW

File: lib/custom_failure.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class CustomFailure < Devise::FailureApp
+  def redirect_url
+     root_path
+  end
+
+  def respond
+    if http_auth?
```
Comment (mdornel) [inline lib/custom_failure.rb]: You could use a shorter syntax `http_auth? ? http_auth : redirect `

**Review by mdornel — CHANGES_REQUESTED** [review]

**Review by mdornel — APPROVED** [review]

File: app/views/users/sessions/create.js.erb [hunk 2]
```diff
@@ -0,0 +1,2 @@
+$('#error-explanation').remove();
```
Comment (gabrielrp) [inline app/views/users/sessions/create.js.erb]: Do you continue using this?

File: spec/controllers/users_controller_spec.rb [hunk 3]
```diff
@@ -30,7 +30,7 @@
       end
 
       it "redirect to sign in page " do
-        expect(response).to redirect_to( new_user_session_path )
+        expect(response).to redirect_to( root_path )
```
Comment (gabrielrp) [inline spec/controllers/users_controller_spec.rb]: Remove spaces between parentheses

File: spec/features/sign_in_spec.rb [hunk 4]
```diff
@@ -32,20 +32,21 @@
...
-      expect(page).to have_current_path(new_user_session_path)
-      expect(page).to have_content( 'Invalid Email or password' )
+      expect(page).to have_current_path(root_path)
     end
 
     scenario 'Invalid Password' do
       page.fill_in 'user_email', with: user.email
       page.fill_in 'user_password', with: '654321'
       click_button  "Sign in"
+      expect(page).to have_content( 'Invalid Email or password' )
```
Comment (gabrielrp) [inline spec/features/sign_in_spec.rb]: Remove spaces

File: spec/features/sign_in_spec.rb [hunk 5]
```diff
@@ -32,20 +32,21 @@
...
     end
 
     scenario 'Invalid Email' do
       page.fill_in 'user_email', with: ''
       page.fill_in 'user_password', with: '123456'
       click_button  "Sign in"
+      expect(page).to have_content( 'You need to sign in or sign up before continuing' )
```
Comment (gabrielrp) [inline spec/features/sign_in_spec.rb]: Same^

**Review by gabrielrp — CHANGES_REQUESTED** [review]

File: spec/features/sign_in_spec.rb [hunk 6]
```diff
@@ -32,20 +32,21 @@
     end
 
     after(:each) do
-      expect(page).to have_current_path(new_user_session_path)
-      expect(page).to have_content( 'Invalid Email or password' )
+      expect(page).to have_current_path(root_path)
```
Comment (gabrielrp) [inline spec/features/sign_in_spec.rb]: This should be an scenario by itself. Always try to have one expect per scenario.

File: spec/features/sign_in_spec.rb [hunk 7]
```diff
@@ -12,17 +12,17 @@
       user.confirm
       page.fill_in 'user_email', with: user.email
       page.fill_in 'user_password', with: '123456'
-      click_button  "Sign in"
+      click_button  'Sign in'
```
Comment (gabrielrp) [inline spec/features/sign_in_spec.rb]: You can create a method `fill_form_and_click` with the email and password as params, so you can reuse the code in all the tests. Something like:
```
def fill_form_and_click(email, password)
  page.fill_in 'user_email', with: email
  page.fill_in 'user_password', with: password
  click_button  'Sign in'
end
```

**Review by gabrielrp — CHANGES_REQUESTED** [review]

File: spec/features/sign_in_spec.rb [hunk 8]
```diff
@@ -32,20 +28,28 @@
     end
 
     after(:each) do
-      expect(page).to have_current_path(new_user_session_path)
-      expect(page).to have_content( 'Invalid Email or password' )
+
```
Comment (gabrielrp) [inline spec/features/sign_in_spec.rb]: Remove the `after` block

File: spec/features/sign_in_spec.rb [hunk 9]
```diff
@@ -10,19 +10,15 @@
   context 'with valid params' do
     scenario 'Sucessful sign in' do
       user.confirm
-      page.fill_in 'user_email', with: user.email
-      page.fill_in 'user_password', with: '123456'
-      click_button  "Sign in"
+      fill_form_and_click user.email, '123456'
       expect(page).to have_current_path(homepage_users_path)
```
Comment (gabrielrp) [inline spec/features/sign_in_spec.rb]: Move this to a new scenario

**Review by gabrielrp — CHANGES_REQUESTED** [review]

**Review by gabrielrp — APPROVED** [review]

**Pull request merged**